anvil/src/shell/ssd.rs

```diff
@@ -0,0 +1,251 @@
+                    Result::<_, ()>::Ok(vec![Rectangle::from_loc_and_size(
+                        ((width - BUTTON_WIDTH) as i32, 0),
+                        (BUTTON_WIDTH as i32, BUTTON_HEIGHT as i32),
+                    )])
+                })
+                .unwrap();
+
+            self.close_button_hover = false;
+        }
+
+        if self
+            .pointer_loc
+            .as_ref()
+            .map(|l| l.x >= (width - BUTTON_WIDTH * 2) as f64 && l.x <= (width - BUTTON_WIDTH) as f64)
+            .unwrap_or(false)
+            && (needs_redraw_buttons || !self.maximize_button_hover)
+        {
+            render_context
+                .draw(|buffer| {
+                    buffer
+                        .chunks_exact_mut((width * 4) as usize)
+                        .flat_map(|x| {
+                            x[((width - (BUTTON_WIDTH * 2)) * 4) as usize
+                                ..((width - BUTTON_WIDTH) * 4) as usize]
+                                .chunks_exact_mut(4)
+                        })
+                        .for_each(|chunk| chunk.copy_from_slice(MAX_COLOR_HOVER));
+                    Result::<_, ()>::Ok(vec![Rectangle::from_loc_and_size(
+                        ((width - BUTTON_WIDTH * 2) as i32, 0),
+                        (BUTTON_WIDTH as i32, BUTTON_HEIGHT as i32),
+                    )])
+                })
+                .unwrap();
+
+            self.maximize_button_hover = true;
+        } else if !self
+            .pointer_loc
+            .as_ref()
+            .map(|l| l.x >= (width - BUTTON_WIDTH * 2) as f64 && l.x <= (width - BUTTON_WIDTH) as f64)
+            .unwrap_or(false)
+            && (needs_redraw_buttons || self.maximize_button_hover)
+        {
+            render_context
+                .draw(|buffer| {
+                    buffer
+                        .chunks_exact_mut((width * 4) as usize)
+                        .flat_map(|x| {
+                            x[((width - (BUTTON_WIDTH * 2)) * 4) as usize
+                                ..((width - BUTTON_WIDTH) * 4) as usize]
+                                .chunks_exact_mut(4)
+                        })
+                        .for_each(|chunk| chunk.copy_from_slice(MAX_COLOR));
+                    Result::<_, ()>::Ok(vec![Rectangle::from_loc_and_size(
+                        ((width - BUTTON_WIDTH * 2) as i32, 0),
+                        (BUTTON_WIDTH as i32, BUTTON_HEIGHT as i32),
+                    )])
+                })
+                .unwrap();
+
+            self.maximize_button_hover = false;
+        }
+    }
+}
+
+impl WindowElement {
+    pub fn decoration_state(&self) -> RefMut<'_, WindowState> {
+        self.user_data().insert_if_missing(|| {
+            RefCell::new(WindowState {
+                is_ssd: false,
+                ptr_entered_window: false,
+                header_bar: HeaderBar {
+                    pointer_loc: None,
+                    width: 128,
```

Comment:
What is the reason behing using `128` as the default `width`? Wouldn't `0` also work as long as the width is not known?